### PR TITLE
Make Volume Plugin FAQ more clear

### PR DESCRIPTION
**Pull request by saad-ali**
The FAQ made assumptions about knowledge that this PR explicitly clarifies.

### CODE REVIEW

File: sig-storage/volume-plugin-faq.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+There are three methods to implement a volume plugin:
+1. In-tree volume plugin
+2. Out-of-tree Flex driver
+3. Out-of-tree CSI driver
+
+The Kubernetes Storage SIG, which is responsible for all volume code in the Kubernetes core repository, is no longer accepting new in-tree volume plugins. Instead, the SIG recommends storage vendors develop plugins as either Flex or CSI drivers.
+
+**What is an in-tree vs out-of-tree volume plugin?**
+
+Before the introduction of the Container Storage Interface (CSI) and FlexVolume, all volume plugins were *in-tree* meaning they were built, linked, compiled, and shipped with the core kubernetes binaries and extend the core Kubernetes API. This meant that adding a new storage system to Kubernetes (a volume plugin) required checking code into the core Kubernetes code repository.
```
Comment (verult) [inline sig-storage/volume-plugin-faq.md]: caps on the K in Kubernetes (first occurence) :)

File: sig-storage/volume-plugin-faq.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+**What happens to existing in-tree volume plugins?**
+
+One of the goals of SIG Storage is to eventually have a CSI-compatible plugin for most existing in-tree plugins and migrate the in-tree plugins to CSI.
+
+## Container Storage Interface (CSI)
+
+**What is the Container Storage Interface (CSI)?**
+
+Container Storage Interface (CSI) is a standardized mechanism for Container Orchestration Systems (COs), including Kubernetes, to expose arbitrary storage systems to containerized workloads. CSI is planned as the primary volume plugin system for Kubernetes. It was introduced in Kubernetes 1.9 as alpha. There are plans to promote it to beta in the 1.10 to 1.11 timeframe.
```
Comment (verult) [inline sig-storage/volume-plugin-faq.md]: CSI is planned **to become** the primary...

File: sig-storage/volume-plugin-faq.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+A *Kubernetes Volume plugin* extends the Kubernetes volume interface to support a block and/or file storage system.
+
+## In-tree vs Out-of-Tree Volume Plugins
+
+**How do I implement a Volume plugin?**
+
+There are three methods to implement a volume plugin:
+1. In-tree volume plugin
+2. Out-of-tree Flex driver
```
Comment (verult) [inline sig-storage/volume-plugin-faq.md]: FlexVolume driver

**Comment by saad-ali** [comment]
Done, PTAL

**Pull request merged**